src/app/metadata/features/metadata-browser/metadata-browser.component.html

```diff
@@ -1,28 +1,66 @@
 <div class="grid grid-cols-12 grid-rows-none gap-5">
   <div class="col-span-9 col-start-4">
+    @if (lastSearchQuery() !== '') {
+      <mat-chip class="float-left"
+        ><button class="align-middle" (click)="clearSearchQuery()">
+          <mat-icon>Close</mat-icon></button
+        >Keyword: {{ lastSearchQuery() }}</mat-chip
+      >
+    }
+    @for (facet of searchResults().facets; track facet.key) {
+      @if (lastSearchFilterFacets()[facet.key]) {
+        @for (option of facet.options; track option.value) {
+          @if (lastSearchFilterFacets()[facet.key].indexOf(option.value) !== -1) {
+            <mat-chip class="float-left">
+              <button
+                class="align-middle"
+                (click)="removeFacet(facet.key + '#' + option.value)"
+              >
+                <mat-icon>Close</mat-icon>
+              </button>
+              {{ facet.name }}: {{ option.value }}
+            </mat-chip>
+          }
+        }
+      }
+    }
     <mat-chip class="float-right">Total Datasets: {{ searchResults().count }}</mat-chip>
   </div>
   <aside class="col-span-3">
     <div class="rounded-lg border p-3">
-      <form>
+      <form (submit)="submit()" #form="ngForm">
```

Comment:
```suggestion
      <form (submit)="submit()">
```
Not sure what this declaration is for?

Reply:
removed